rasa/utils/tensorflow/model_data_utils.py

```diff
@@ -30,6 +30,7 @@ def featurize_training_examples(
     entity_tag_specs: Optional[List["EntityTagSpec"]] = None,
     featurizers: Optional[List[Text]] = None,
     bilou_tagging: bool = False,
+    type: Optional[Text] = None,

```

Comment:
I think a better name would be `feature_type` since `type` could refer to sparse or dense features also. Wdyt?

Reply:
I think that we should keep this type because type is what is used elsewhere. In long term I'd vote for calling this "level" and denote sparse/dense as "type" :D.

Reply:
I can see that you have already used "level" in some parts of the code. (nitpick: Even there "feature_level" is better than generic "level" IMO but no hard opinions and an optional suggestion). If we call it `type` here and `level` somewhere else then we are increasing more discrepancy. :)

Reply:
Yeah, no I would not call it level here now. I had called it that in an intermediate version and tried to call sparse/dense "type" but that looks very messy/confusing then without refactoring everything else.  
Will keep "type" here, if you're not strongly against it. 

> I can see that you have already used "level" in some parts of the code
but only as in "sentence-level" features or "sequence-level" features, which sounds ok, no? Or is there a variable I overlooked?

Reply:
found some more "lvl" and "level" and replaced them, let me know if you see more

Reply:
Okay for now. In the future `feature_type` and `feature_level` make sense to me.